<launch>

  <!-- <arg name="input_mesh" default="/infinitam/mesh"/> -->
  <arg name="params_file" default="$(find mochapc)/learning_params.csv"/>
  <arg name="mode" default="0"/>
  <arg name="terrain_mesh_file" default="$(find mochapc)/labLoop.ply"/>
  <arg name="car_mesh_file" default="$(find mochapc)/herbie/herbie.blend"/>
  <arg name="wheel_mesh_file" default="$(find mochapc)/herbie/wheel.blend"/>
  <arg name="terrain_mesh_topic" default="vehicle/input_terrain_mesh"/>
  <arg name="publish_timeout" default="1.0"/>

  <node pkg="nodelet" type="nodelet" name="mocha_manager"  args="manager"/>
  
  
  <node pkg="nodelet" type="nodelet" name="static_mesh_publisher" args="load mochapc/StaticMeshPublisherNodelet mocha_manager">
    <param name="terrain_mesh_file" value="$(arg terrain_mesh_file)"/>
    <param name="terrain_mesh_topic" value="$(arg terrain_mesh_topic)"/>
    <param name="publish_timeout" value="$(arg publish_timeout)"/>
  </node>
  <node pkg="nodelet" type="nodelet" name="mocha_vehicle" args="load mochapc/MochaVehicleNodelet mocha_manager">
    <param name="params_file" value="$(arg params_file)"/>
    <param name="mode" value="$(arg mode)"/>
    <param name="terrain_mesh_file" value="$(arg terrain_mesh_file)"/>
    <param name="car_mesh_file" value="$(arg car_mesh_file)"/>
    <param name="wheel_mesh_file" value="$(arg wheel_mesh_file)"/>
  </node>

</launch>
